### PR TITLE
Added loading indicator to show progress when delay in showing option

### LINKED ISSUE #928: Sign-in page - delay in showing authenticator options
**Description**:
Reported by Franz

  -	Sometimes the login page would only show the lock symbol and “Sign in” but not the actual fields for the credentials until quite a wait:
 
 
signin

**Acceptance criteria**:
- [ ] Check what is causing delays in showing authenticators and display a message if this happens

### PR DESCRIPTION
**Pull request by AkhilMag13**
## Description
Added a loading indicator to show progress while the authenticator options are not yet loaded.

## Testing instructions
Throttling the internet connection to slow 3G to see effect
- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
Closes #928

### CODE REVIEW

**Comment by joshuadkitenge** [comment]
login loading indicator

File: src/loginPage/loginPage.component.tsx [hunk 1]
```diff
@@ -416,6 +426,7 @@ const LoginPageComponent = (props: CombinedLoginProps): React.ReactElement => {
   const authUrl = props.auth.provider.authUrl;
   const [mnemonics, setMnemonics] = useState<ICATAuthenticator[]>([]);
   const [fetchedMnemonics, setFetchedMnemonics] = useState<boolean>(false);
+  const [isComponentVisible, setComponentVisible] = useState<boolean>(false);
```
Comment (louise-davies) [inline src/loginPage/loginPage.component.tsx:429]: Slight comment that could this be renamed to be something like `loading` and `setLoading` - you'd have to reverse the logic (or use `loaded` and `setLoaded` I suppose...) - `isComponentVisisble` is a bit confusing

**Review by louise-davies — CHANGES_REQUESTED** [review]
Rather than pass `setComponentVisible` to a bunch of components that don't really need it (i.e. they don't have any sort of loading), I think we can instead remove it by having the loading component run on the same logic as what triggers the mnemonic fetching in the first place i.e

```
{typeof mnemonic !== 'undefined' && !fetchedMnemonics ? (
    <CircularProgress className={props.classes.spinner} />
) : null}
```

Also, we don't need to different `CircularProgress` components - we can just use or conditions to have it render when either `props.auth.loading || typeof mnemonic !== 'undefined' && !fetchedMnemonics`

File: src/loginPage/loginPage.component.tsx [hunk 2]
```diff
@@ -551,6 +553,10 @@ const LoginPageComponent = (props: CombinedLoginProps): React.ReactElement => {
         {props.auth.loading ? (
           <CircularProgress className={props.classes.spinner} />
         ) : null}
+
+        {typeof mnemonic !== 'undefined' && !fetchedMnemonics ? (
+          <CircularProgress className={props.classes.spinner} />
+        ) : null}
```
Comment (louise-davies) [inline src/loginPage/loginPage.component.tsx:559]: These could still be incorporated into one conditional - I don't know if that's more or less readable

Comment (joshuadkitenge) [reply src/loginPage/loginPage.component.tsx]: It's still readable. I have added the changes

**Review by louise-davies — APPROVED** [review]

**Pull request merged**